src/proofs/Term.cpp

```diff
@@ -71,7 +71,7 @@ std::string Op::simplifyRule() {
     std::string op = operation;
     if (op == "=") {
         if ((args[0]->printTerm().find_first_not_of("( )-0123456789") == std::string::npos) and
-            (args[0]->printTerm().find_first_not_of("( )-0123456789") == std::string::npos)) {
+            (args[1]->printTerm().find_first_not_of("( )-0123456789") == std::string::npos)) {
```

Comment:
Wait, so this was a bug?
Can you come up with a test case that would expose it in the previous version?

Reply:
Yes, it probably got changed after I modified the print function. Here I am basically checking if both arguments of the equality are numbers. Hence the index 0 and 1.